TheraLang.Web/ClientApp/src/app/core/http/project/project.service.ts

```diff
@@ -64,6 +66,13 @@ export class ProjectService {
   }
 
   addProject(project: Project) {
+    let token: string = localStorage.getItem("jwt");
+    if (token && !this.jwtHelper.isTokenExpired(token)) {
+      let jwtData = this.jwtHelper.decodeToken(token);
+      let userId = jwtData['Id']
+        project.createdbyid=userId;
```

Comment:
or assign createdById value on backend?

Reply:
let's leave it as it is

Reply:
On a server you could check if user with given id exist. So better move it

Reply:
I dont check if user exists. When a user creates a project, a viewmodel is sent to the backend. I think it is logical to fill in all the fields of this model on the front and just pass to the backend, as opposed to filling some fields in the front and some in the backend. So, I pass the id of the user who created this project along with the other fields filled in to the backend. Is this a problem? If this is not good practice, I will move it.